tests/mocha/workspace_test.js

```diff
@@ -120,66 +133,48 @@ suite('Workspace', function() {
-      var blockVarName2 = this.workspace.topBlocks_[1].getVarModels()[0].name;
-      chai.assert.equal(blockVarName2, 'name1');
-      var blockVarName3 = this.workspace.topBlocks_[2].getVarModels()[0].name;
-      chai.assert.equal(blockVarName3, 'name2');
+      assertBlockVarModelName(this.workspace, 0, 'name1');
+      assertBlockVarModelName(this.workspace, 1, 'name1');
+      assertBlockVarModelName(this.workspace, 2, 'name2');
     });
   });
 
-  suite('renameVariable', function() {
+  suite('renameVariableById', function() {
     setup(function() {
       this.workspace.createVariable('name1', 'type1', 'id1');
     });
-    test('No references', function() {
+
+    test('No references rename to name2', function() {
       this.workspace.renameVariableById('id1', 'name2');
       assertVariableValues(this.workspace, 'name2', 'type1', 'id1');
       // Renaming should not have created a new variable.
       chai.assert.equal(this.workspace.getAllVariables().length, 1);
     });
 
-    test('Reference exists', function() {
-      // Turn off events to avoid testing XML at the same time.
-      Blockly.Events.disable();
-      var block = new Blockly.Block(this.workspace, 'get_var_block');
-      block.inputList[0].fieldRow[0].setValue('id1');
-      Blockly.Events.enable();
+    test('Reference exists rename to name2', function() {
+      createVarBlocksNoEvents(this.workspace, ['id1']);
 
       this.workspace.renameVariableById('id1', 'name2');
       assertVariableValues(this.workspace, 'name2', 'type1', 'id1');
       // Renaming should not have created a new variable.
       chai.assert.equal(this.workspace.getAllVariables().length, 1);
-      var blockVarName = this.workspace.topBlocks_[0].getVarModels()[0].name;
-      chai.assert.equal(blockVarName, 'name2');
+      assertBlockVarModelName(this.workspace, 0, 'name2');
     });
 
-    test('Reference exists different capitalizations', function() {
-      // Turn off events to avoid testing XML at the same time.
-      Blockly.Events.disable();
-      var block = new Blockly.Block(this.workspace, 'get_var_block');
-      block.inputList[0].fieldRow[0].setValue('id1');
-      Blockly.Events.enable();
+    test('Reference exists different capitalization rename to Name1', function() {
+      createVarBlocksNoEvents(this.workspace, ['id1']);
 
       this.workspace.renameVariableById('id1', 'Name1');
       assertVariableValues(this.workspace, 'Name1', 'type1', 'id1');
       // Renaming should not have created a new variable.
       chai.assert.equal(this.workspace.getAllVariables().length, 1);
-      var blockVarName = this.workspace.topBlocks_[0].getVarModels()[0].name;
-      chai.assert.equal(blockVarName, 'Name1');
+      assertBlockVarModelName(this.workspace, 0, 'Name1');
     });
 
     suite('Two variables rename overlap', function() {
-      test('Same type', function() {
+      test('Same type rename variable id=id1 to name2', function() {
```

Comment:
Maybe just 'rename variable with id1 to name2'? Don't know why but the id= threw me off a bit.

Reply:
Sure. I've been tossing up a few variations, initially I had rename id1 to name2, but that seemed like it could be misunderstood as renaming something with initial name as id1.